### PR TITLE
Improve ens/zns coding style

### LINKED ISSUE #491: Improve ens.go coding style
- [ ] return `blockatlas.Resolved` instead of `*blockatlas.Resolved`
- [ ] get `ethereum.rpc` url from `Platform` not from `viper` directly
- [ ] return error if not nil

### PR DESCRIPTION
**Pull request by hewigovens**
Fix #491

### CODE REVIEW

File: pkg/blockatlas/api.go [hunk 1]
```diff
@@ -70,5 +70,5 @@ type CustomAPI interface {
 
 type NamingServiceAPI interface {
 	Platform
-	Lookup(coin uint64, name string) (*Resolved, error)
+	Lookup(coin uint64, name string) Resolved
```
Comment (vikmeup) [inline pkg/blockatlas/api.go]: I would suggest to keep `(Resolved, error)` - this is a proper way to handle it

File: platform/ethereum/ens.go [hunk 2]
```diff
@@ -38,33 +31,32 @@ func (p *Platform) Lookup(coin uint64, name string) (*blockatlas.Resolved, error
...
 
-func lookupLegacyETH(client *ethclient.Client, name string) (*blockatlas.Resolved, error) {
+func lookupLegacyETH(client *ethclient.Client, name string) blockatlas.Resolved {
 	result := blockatlas.Resolved{
 		Coin: CoinType.ETH,
 	}
 	resolver, err := ens.NewResolver(client, name)
 	if err != nil {
-		return &result, nil
+		result.Error = errors.Wrap(err, "new ens resolver failed").Error()
```
Comment (Pantani) [inline platform/ethereum/ens.go]:
```suggestion
		result.Error = errors.E(err, "new ens resolver failed")
```

File: platform/ethereum/ens.go [hunk 3]
```diff
@@ -38,33 +31,32 @@ func (p *Platform) Lookup(coin uint64, name string) (*blockatlas.Resolved, error
...
+		return result
 	}
 	address, err := resolver.Address()
 	if err != nil {
-		return &result, nil
+		result.Error = errors.Wrap(err, "query address failed").Error()
```
Comment (Pantani) [inline platform/ethereum/ens.go]:
```suggestion
		result.Error = errors.E(err, "query address failed")
```

File: platform/zilliqa/api.go [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
-	"strconv"
 
 	"github.com/spf13/viper"
 )
 
 type Platform struct {
 	client    Client
 	rpcClient RpcClient
+	udClient  blockatlas.Request
```
Comment (Pantani) [inline platform/zilliqa/api.go]: It's better to extend, and you can use more simple

```suggestion
	blockatlas.Request
```

`err := p.udClient.Get(&resp, "/"+name, nil)`

File: platform/ethereum/ens.go [hunk 5]
```diff
@@ -38,33 +31,32 @@ func (p *Platform) Lookup(coin uint64, name string) (*blockatlas.Resolved, error
...
 	}
 
 	encoded, err := cc.ToString(address, uint32(coin))
 	result.Result = encoded
 	if err != nil {
-		logger.Error("Encode to address failed", err, logger.Params{
-			"address": hex.EncodeToString(address),
-			"coin":    coin,
-		})
+		result.Error = errors.Wrap(err, "encode to address failed").Error()
```
Comment (Pantani) [inline platform/ethereum/ens.go]:
```suggestion
		result.Error = errors.E(err, "encode to address failed")
```

File: platform/ethereum/ens.go [hunk 6]
```diff
@@ -1,35 +1,28 @@
...
-func (p *Platform) Lookup(coin uint64, name string) (*blockatlas.Resolved, error) {
-	client, err := ethclient.Dial(viper.GetString("ethereum.rpc"))
+func (p *Platform) Lookup(coin uint64, name string) blockatlas.Resolved {
+	client, err := ethclient.Dial(p.RpcURL)
 	result := blockatlas.Resolved{
 		Coin: coin,
 	}
 	if err != nil {
-		return nil, errors.E(http.StatusInternalServerError, "can't dial to ethereum rpc")
+		result.Error = errors.Wrap(err, "can't dial to ethereum rpc").Error()
```
Comment (Pantani) [inline platform/ethereum/ens.go]: Why do you remove our error package?

**Comment by hewigovens** [comment]
@vikmeup @Pantani take another look

**Pull request merged**